### PR TITLE
Add Spotbugs to Maven and Fix Spotbugs Warnings

### PR DESCRIPTION
**Pull request by OwenCorrigan76**
Spotbugs added to Client Examples. As a result of Spotbug warnings, I have edited the files with warnings as best as i could to get rid of the warnings but ensure that the code still works properly. Each component now compiles and runs as expected and has no Spotbugs warnings. I suspect there will be some comments on how my changes may have been carried out more efficiently.

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
This looks good. But you should also add it to the CI in https://github.com/strimzi/client-examples/tree/main/.azure -> so that it is actually executed in the CI pipeline.

**Comment by OwenCorrigan76** [comment]
@scholzj I've added the following to the `build-pipeline.yaml` under jobs/ steps/ 
```
- script: 'make spotbugs'
  displayName: 'Run SpotBugs' 
```
Is this sufficient or do i need to add anything else in the Azure directory? Thanks

**Comment by scholzj** [comment]
@OwenCorrigan76 I think that might be fine -> push it in and we will see.

File: pom.xml [hunk 1]
```diff
@@ -26,9 +26,12 @@
         <maven.shade.version>3.4.1</maven.shade.version>
         <maven.checkstyle.version>3.2.0</maven.checkstyle.version>
 
+
+
```
Comment (scholzj) [inline pom.xml]: Not needed I guess?

File: python/kafka/producer/__pycache__/kafka_producer_config.cpython-310.pyc
Comment (scholzj) [inline python/kafka/producer/__pycache__/kafka_producer_config.cpython-310.pyc:1]: This file probably should not be there?

**Review by scholzj — APPROVED** [review]
Two more nits to fix. But LGTM otherwise.

File: pom.xml (hunk 1 above)
Comment (OwenCorrigan76) [reply pom.xml]: Extra lines removed

File: python/kafka/producer/__pycache__/kafka_producer_config.cpython-310.pyc
Comment (OwenCorrigan76) [reply python/kafka/producer/__pycache__/kafka_producer_config.cpython-310.pyc:1]: Removed

**Review by scholzj — APPROVED** [review]

File: java/kafka/pom.xml [hunk 2]
```diff
@@ -18,4 +18,16 @@
...
+    <plugin>
+        <groupId>com.github.spotbugs</groupId>
+        <artifactId>spotbugs-maven-plugin</artifactId>
+        <configuration>
+            <!-- Configures the file for excluding warnings -->
+            <excludeFilterFile>${project.basedir}/../../../.spotbugs/spotbugs-exclude.xml</excludeFilterFile>
+        </configuration>
+    </plugin>
+    </plugins>
+    </build>
```
Comment (ppatierno) [inline java/kafka/pom.xml:32]: wrong indentation of plugins and plugin tags

**Review by ppatierno — APPROVED** [review]
Just one nit but LGTM. No need to re-review when it's ok. We can just merge.

File: java/kafka/pom.xml (hunk 2 above)
Comment (OwenCorrigan76) [reply java/kafka/pom.xml:32]: @ppatierno Indentation fixed

**Pull request merged**